packages/php-wasm/universal/src/lib/php-request-handler.ts

```diff
@@ -313,7 +316,7 @@ export class PHPRequestHandler<PHP extends BasePHP> {
 			this.rewriteRules
 		);
 		const fsPath = joinPaths(this.#DOCROOT, normalizedRequestedPath);
-		if (!seemsLikeAPHPRequestHandlerPath(fsPath)) {
+		if (!(await this.#seemsLikeAPHPRequestHandlerPath(fsPath))) {
```

Comment:
At this point I'd refactor the routing logic completely. We can resolve the URL to something like `{ "filePath": string, "isRemote": boolean }` with 100% certainty and then test whether `filePath` is a file with `.php` extension. If the requested path is `/index.php/a/b/c` then it could mean different things, e.g. `I'm requesting a static file called "c" that's inside a directory tree like $DOCROOT/index.php/a/b/`, or `I'm requesting index.php and /a/b/c is my "nice URL"`. Let's do what a web server would do here.

Reply:
If we do a full refactor let's also take this [use case into account. ](https://github.com/WordPress/wordpress-playground/pull/1187), I [started working on it,](https://github.com/WordPress/wordpress-playground/pull/1203) but lacked the metadata to complete it.

Reply:
> Let's do what a web server would do here.

Right. Let's get it done. I'll work on that.

> If we do a full refactor let's also take this use case into account. 

Got it. Hopefully that will just naturally work after the refactor, but we can test it.